src/annotation/decorator_utils.ts

```diff
@@ -1,68 +1,110 @@
 import * as ERROR_MSGS from "../constants/error_msgs";
 import * as METADATA_KEY from "../constants/metadata_keys";
 import { interfaces } from "../interfaces/interfaces";
+import { getFirstArrayDuplicate } from "../utils/js";
+
+function targetIsConstructorFunction<T = Object>(target:DecoratorTarget<T>): target is ConstructorFunction<T>{
```

Comment:
```suggestion
function targetIsConstructorFunction<T = Object>(target: unknown): target is ConstructorFunction<T>{
```

Reply:
At this point, are we sure target is a DecoratorTarget<T>? If that's the case we can avoid this

Reply:
Why ?

Reply:
It is known and it certainly is not undefined or null

Reply:
> At this point, are we sure target is a DecoratorTarget? If that's the case we can avoid this

Avoid the change to unknown ?

We are sure it is a decorator target as the decorator is called with the decorator target.  tagProperty is called with the decorator target and passes it to the type predicate